cumulus_library/base_table_builder.py

```diff
@@ -32,14 +34,31 @@ def prepare_queries(self, cursor: object, schema: str):
         raise NotImplementedError
 
     @final
-    def execute_queries(self, cursor: object, schema: str, verbose: bool):
+    def execute_queries(
+        self, cursor: object, schema: str, verbose: bool, drop_table: bool = False
+    ):
         """Executes queries set up by a prepare_queries call
 
         :param cursor: A PEP-249 compatible cursor
         :param schema: A schema name
         :param verbose: toggle for verbose output mode
+        :param drop_table: drops any tables found in prepared_queries results
         """
         self.prepare_queries(cursor, schema)
+        if drop_table:
+            table_names = []
+            for query in self.queries:
+                # Get the first non-whitespace word after create table
```

Comment:
I'm sure this is the most reasonable box we could put ourselves into, but all this manual, regex-based SQL parsing gives me the heebie-jeebies. At least sqlfluff makes our own SQL fairly regular, right?

Maybe it's not worth the effort, but is there an actual SQL parser we could import & use to tokenize the string? That still leaves us with assuming some SQL structure, but it's one less assumption-about-input. (This is all a nit suggestion)

Reply:
so - i hear you, this is a little fast and dirty. here is my thought:
- We are only using the table creator to, well, create tables
- It's case insensitive

but theoretically if you're doing something besides creating a table, we could modify this to do something like 'if the number of matches is just one, continue without trying to drop tables'.

I'm kind of willing to live with it being a bit risky because this is really just a development tool for iterating on a builder without commenting things out. That said, there are sql parsers we could try to use for this - lemme spend 15 mins looking into it.

Reply:
python-sqlparse is very unpleasant - i'm leaving the regex, but will check to see if the regex passes.